includes/Checker/Checks/Plugin_Readme_Check.php

```diff
@@ -217,16 +272,38 @@ private function check_for_warnings( Check_Result $result, string $readme_file,
 		$warning_keys = array_diff( $warning_keys, $ignored_warnings );
 
 		if ( ! empty( $warning_keys ) ) {
-			$this->add_result_warning_for_file(
-				$result,
-				sprintf(
-					/* translators: list of warnings */
-					esc_html__( 'The following readme parser warnings were detected: %s', 'plugin-check' ),
-					esc_html( implode( ', ', $warning_keys ) )
-				),
-				'readme_parser_warnings',
-				$readme_file
-			);
+			foreach ( $warning_keys as $warning ) {
+				$this->add_result_warning_for_file( $result, $messages[ $warning ], 'readme_parser_warnings', $readme_file );
+			}
+		}
+	}
+
+	/**
+	 * Returns current major WordPress version.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Stable WordPress version.
+	 */
+	private function get_wordpress_stable_version() {
+		$version = get_transient( 'wp_plugin_check_latest_wp_version' );
+
+		if ( false === $version ) {
+			$response = wp_remote_get( 'https://api.wordpress.org/core/version-check/1.7/' );
+
+			if ( ! is_wp_error( $response ) ) {
+				$body = json_decode( wp_remote_retrieve_body( $response ), true );
+
+				if ( isset( $body['offers'] ) && ! empty( $body['offers'] ) ) {
+					$latest_release = reset( $body['offers'] );
+
+					$version = $latest_release['new_bundled'];
+
+					set_transient( 'wp_plugin_check_latest_wp_version', $version, DAY_IN_SECONDS );
+				}
+			}
 		}
+
+		return $version;
```

Comment:
If the HTTP request failed, this will still be false. Maybe we should keep the bloginfo logic as a fallback?